### PR TITLE
Check for additionalProperties to be set to false

### PR DESCRIPTION
**Pull request by shapirone**
###  Summary

The backend defaults `additionalProperties` on parameters of the `object` type to true, so this PR inverts the check to see if they are explicitly setting `additionalProperties` to false.

### Testing
1. Create an `object` parameter as a function input or output
2. Verify that at both workflow authoring time and at runtime, you're able to access unexpected properties.
3. Change the object parameters you created to set `additionalProperties: false`
4. Notice that your access of unexpected properties now results in errors.

```ts
anObject: {
        type: Schema.types.object,
        properties: {
            test: { type: Schema.types.string },
        },
        additionalProperties: false,
      },
```

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/{project_slug}/blob/main/.github/CONTRIBUTING.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: src/functions/interactivity/block_actions_types.ts [hunk 1]
```diff
@@ -76,6 +76,7 @@ export type BlockActionsBody = {
     id: string;
     /**
      * @description User's handle as seen in the Slack client when e.g. at-notifying the user.
+     */
```
Comment (shapirone) [inline src/functions/interactivity/block_actions_types.ts:79]: There was an issue here where we weren't closing the comment, which meant that `name` wasn't part of the BlockActionsBody defined payload. This is unrelated to the core of this PR and can be removed if we want.

File: src/functions/slack_function_handler_type_test.ts [hunk 2]
```diff
@@ -371,3 +371,125 @@ Deno.test("SlackFunctionHandler using Custom Types", () => {
...
+  };
+
+  const { createContext } = SlackFunctionTester(TestFunction);
+
+  const result = handler(createContext({ inputs: sharedInputs }));
+  assertEqualsTypedValues(sharedInputs, result.outputs);
+  assertExists(result.outputs?.noAddlPropertiesObj);
+  assertExists(result.outputs?.noAddlPropertiesObj.aString);
+
+  // @ts-expect-error anythingElse cant exist
```
Comment (shapirone) [inline src/functions/slack_function_handler_type_test.ts:493]: I'm not sure there's a better way to do this type of test, but if this directive becomes unused (i.e. no error), it will let us know!

File: src/functions/types.ts [hunk 3]
```diff
@@ -93,15 +93,20 @@ type UnknownRuntimeType = any;
...
   }
-  & {
-    [key: string]: UnknownRuntimeType;
-  };
+  : 
+    & {
+      [k in keyof Param["properties"]]: FunctionInputRuntimeType<
+        Param["properties"][k]
+      >;
+    }
```
Comment (shapirone) [inline src/functions/types.ts:106]: We could probably figure out some way to avoid this duplication, but I didn't want to spend too much time on this. Open to suggestions.

File: src/functions/slack_function_handler_type_test.ts (hunk 2 above)
Comment (selfcontained) [reply src/functions/slack_function_handler_type_test.ts:493]: TIL

**Review by selfcontained — APPROVED** [review]
beautiful, works as described.
